src/decoder.rs

```diff
@@ -96,8 +96,9 @@ impl Decoder {
     }
 
     pub fn decode_raw<M: Memory>(&mut self, memory: &mut M, pc: u64) -> Result<Instruction, Error> {
-        // since we are using RISCV_MAX_MEMORY as the default key in the instruction cache, have to check out of bound error first
-        if pc as usize >= RISCV_MAX_MEMORY {
+        // since we are using RISCV_MAX_MEMORY as the default key in the instruction cache, have to check out of bound
+        // error first.
+        if pc as usize >= memory.memory_size() {
```

Comment:
Notice when are creating a decoder, RISCV_MAX_MEMORY is still being used above.

Reply:
I noticed.

No bugs at the moment, but maybe u64::MAX is better.